src/model/Sunflower.java

```diff
@@ -1,36 +1,47 @@
-package model;
-/* Sunflower Class is an essential entity for the game.
- * Sunflower generates cashflow for the player to buy new plants.*/
-public class Sunflower extends Plants {
-
-	private final int SUN_POINTS_GENERATED = 3;
-
-	public Sunflower() {
-		super(10, 8, 2);
-	}
-
-	/*
-	 * sunFlower generates sun points based on the turnsUntilAbility variable. ex/
-	 * turnsUntilAbility = 2, every two turns the sunFower will generate extra sun
-	 * points
-	 */
-	public int generateSunPoints() {
-		this.turnsUntilAbility--;
-
-		if (this.turnsUntilAbility == 0) {
-			this.turnsUntilAbility = this.abilityFrequency;
-			return SUN_POINTS_GENERATED;
-		}
-		return 0;
-	}
-
-	@Override
-	public String toString() {
-		return PlantTypes.SUNFLOWER.getName();
-	}
-
-	@Override
-	public char getType() {
-		return PlantTypes.SUNFLOWER.getType();
-	}
-}
+package model;
+
+import model.Stats;
+
+/* Sunflower Class is an essential entity for the game.
+ * Sunflower generates cashflow for the player to buy new plants.*/
+public class Sunflower extends Plants {
+
+	private final int SUN_POINTS_GENERATED = 3;
+    private static int sunPointCost = 8;
+    private Stats stats;
+
+	public Sunflower() {
+		super(10, 8, 2);
+        this.stats = Stats.getStats();
+        this.stats.increaseSunPointsGenerationRate(SUN_POINTS_GENERATED);
+	}
+
+    public static int getCost() {
+        return Sunflower.sunPointCost;
```

Comment:
What's this for? Sunflower already inherits a similar method

Reply:
Sorry, I meant to explain this in my PR but I had to go to class. In [`BoardController`](https://github.com/vikrambombhi/plants-vs-zombies/pull/23/files#diff-4fda040e355b4fd41edce3b1bb397f19R35) I get the cost of the plant **before** creating a instace of it. This is because now when you create a Sunflower it also updates the Stats on how many sunpoints are gained per turn.